### ISSUE TITLE
Synthetic method is not saved

### ISSUE THREAD

**Issue by bagipro** [skylot/jadx#361]
Hi, I've noticed that in the APK (https://drive.google.com/file/d/1LGsMgNOyo9SmZ8FEBPYDWKWy4hrtTmPS/view?usp=sharing) a private method in class ``` com.kayak.android.Splash ``` 
```java
    private com.kayak.android.LaunchMode determineLaunchMode(java.lang.String str, java.lang.String str2, java.lang.String str3) {
        java.lang.String str4;
        java.lang.String action = getIntent().getAction();
        //...
```
exists, but there are no any references to that method. I used apktool to look what happens and noticed that it's invoked in another synthetic method
```smali
.method final synthetic a(Ljava/util/List;Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Lcom/kayak/android/o;
    .locals 0
    .annotation system Ldalvik/annotation/Throws;
        value = {
            Ljava/lang/Exception;
        }
    .end annotation

    .line 209
    invoke-direct {p0, p2, p3, p4}, Lcom/kayak/android/Splash;->determineLaunchMode(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Lcom/kayak/android/o;

    move-result-object p1

    return-object p1
.end method
```
which is missed in decompiled sources. Can you please fix which synthetic methods are ignored?

**Comment by sergey-wowwow** [comment]
@skylot may you please look at this one? I've noticed similar issue in different app

**Comment by skylot** [comment]
@sergey-wowwow I made a small fix, this was broken by #336 :(
